### DISCUSSION TITLE
Vehicle Routing Problem -- equal utilization of all vehicles

### DISCUSSION THREAD

**Discussion by rahulj2409** [google/or-tools#2640] [Routing (and legacy CP) questions]
Python

For the Vehicle Routing problem in Google-OR, it seems that you must put in this value: # vehicle maximum travel distance when adding the distance dimension. Is there a way to ignore this value for now? I have been manually tuning it to different results IE too high of a value results in less vehicles being used and too low of a value results in no solution being found.

Furthermore,  I was wondering if there's a way for me to ignore putting in a "Vehicle maximum travel distance", but rather have it be that the solution returned by google OR utilizes all of my vehicles "as equally as possible."

Thanks!

**Comment by Mizux** [comment]
1. Make a large enough value so it won't be a limiting factor.

2. Create a counter dimension and use CumulVarSoftLower/UpperBound()...
e.g. 
```python
 # Try to equaly distribute visit among the fleet
    routing.AddVectorDimension(
        data['visit_counter'],
        manager.GetNumberOfNodes(),
        True,  # start cumul to zero
        "Counter")
    counter_dimension = routing.GetDimensionOrDie("Counter")
    num_visit = sum(data['visit_counter']) // manager.GetNumberOfVehicles()
    print(f'num of visit per vehicle: {num_visit}')

    for vehicle_id in range(data["num_vehicles"]):
        index = routing.End(vehicle_id)
        counter_dimension.SetCumulVarSoftLowerBound(index, num_visit, 100_000)
        counter_dimension.SetCumulVarSoftUpperBound(index, num_visit + 1,
                                                    100_000)
```

**Reply by rahulj2409** [reply]
What is data['visit_counter']?

**Reply by Mizux** [reply]
an array with 1 for location, and 0 for depot aka count +1 per location but take into account start/end node etc...
`data['visit_counter'] = [0, 1, 1, ....]`
